examples/cockroachdb/cockroachdb-statefulset.yaml

```diff
@@ -66,6 +78,21 @@ spec:
       labels:
         app: cockroachdb
       annotations:
+        scheduler.alpha.kubernetes.io/affinity: >
```

Comment:
It actually doesn't appear to be getting strictly enforced :/ Is there an existing bug around this?

I had initially tried this on minikube (1 node, 1.5.1) and all the pods were scheduled/started just fine. I also just tried it on a 3 node cluster on GKE (master on 1.5.1, nodes on 1.4.7) and found that it put two of the replicas on the same node even when there were only 3 replicas. Even worse, there was sufficient space to have put one of the replicas on the third node, given that when I scaled up to 4 pods the fourth got placed on the third node.